### PR TITLE
feat: Show permissions boundary policy name in app show

### LINKED ISSUE #4064: Show permissions boundary policy name in app show
Now that we have support for permissions boundary for apps created through copilot, we can very well display this detail on app show command.

```
About

  Name                  fifobug
  Version               v1.1.0 (latest available: v1.0.2)
  URI      
  Permissions Boundary  MyCopilotPermissionBoundary
```

### PR DESCRIPTION
**Pull request by KollaAdithya**
This PR resolves #4064

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/describe/app.go [hunk 1]
```diff
@@ -23,14 +23,15 @@ import (
...
-	URI                string                   `json:"uri"`
-	Envs               []*config.Environment    `json:"environments"`
-	Services           []*config.Workload       `json:"services"`
-	Jobs               []*config.Workload       `json:"jobs"`
-	Pipelines          []*codepipeline.Pipeline `json:"pipelines"`
-	WkldDeployedtoEnvs map[string][]string      `json:"-"`
+	Name                string                   `json:"name"`
+	Version             string                   `json:"version"`
+	URI                 string                   `json:"uri"`
+	PermissionsBoundary string                   `json:"permissionsboundary"`
```
Comment (paragbhingre) [inline internal/pkg/describe/app.go]: nit:
```suggestion
	PermissionsBoundary string                   `json:"permissionsBoundary"`
```

File: internal/pkg/cli/app_show_test.go [hunk 2]
```diff
@@ -347,9 +351,10 @@ Pipelines
 
 			wantedContent: `About
 
-  Name     my-app
-  Version  v1.1.0 
-  URI      example.com
+  Name                  my-app
+  Version               v1.1.0 
+  URI                   example.com
+  Permissions Boundary  examplePermissionsBoundaryPolicy
```
Comment (paragbhingre) [inline internal/pkg/cli/app_show_test.go:357]: We can also have one test case without having permissions boundary

File: internal/pkg/describe/app.go [hunk 3]
```diff
@@ -55,6 +56,7 @@ func (a *App) HumanString() string {
 	}
 	fmt.Fprintf(writer, "  %s\t%s %s\n", "Version", a.Version, availableVersion)
 	fmt.Fprintf(writer, "  %s\t%s\n", "URI", a.URI)
+	fmt.Fprintf(writer, "  %s\t%s\n", "Permissions Boundary", a.PermissionsBoundary)
```
Comment (efekarakus) [inline internal/pkg/describe/app.go:65]: nit: I wonder if we should write "N/A" if there is no setting for these fields what do you think?
So that it looks like this when empty:
```
About

  Name                  demo
  Version               v1.1.0 (latest available: v1.0.2)
  URI                   N/A
  Permissions Boundary  N/A
```

**Review by efekarakus — APPROVED** [review]
Looks great to me! Just a tiny UX suggestion and +1 to Parag's suggestion

**Review by huanjani — APPROVED** [review]
This looks great! Thank you. It'll be so helpful to surface this info to users!

File: internal/pkg/describe/app.go [hunk 4]
```diff
@@ -53,8 +54,15 @@ func (a *App) HumanString() string {
...
 		availableVersion = color.Yellow.Sprintf("(latest available: %s)", deploy.LatestAppTemplateVersion)
 	}
+	if a.URI == "" {
+		a.URI = "N/A"
+	}
+	if a.PermissionsBoundary == "" {
+		a.PermissionsBoundary = "N/A"
+	}
 	fmt.Fprintf(writer, "  %s\t%s %s\n", "Version", a.Version, availableVersion)
 	fmt.Fprintf(writer, "  %s\t%s\n", "URI", a.URI)
```
Comment (efekarakus) [inline internal/pkg/describe/app.go:64]: Unrelated to your PR but I feel like we should rename this from "URI" to "Domain" in a separate PR

**Review by efekarakus — APPROVED** [review]
